server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -498,15 +502,23 @@ public void updateCachedBlockIds(
     }
     ShuffleTaskInfo shuffleTaskInfo =
         shuffleTaskInfos.computeIfAbsent(appId, x -> new ShuffleTaskInfo(appId));
-    Roaring64NavigableMap bitmap =
-        shuffleTaskInfo
-            .getCachedBlockIds()
-            .computeIfAbsent(shuffleId, x -> Roaring64NavigableMap.bitmapOf());
-
     long size = 0L;
-    synchronized (bitmap) {
+    // With memory storage type should never need cachedBlockIds,
+    // since client do not need call finish shuffle rpc
+    if (!shuffleTaskInfo.isClientStorageTypeWithMemory()) {
```

Comment:
We can get this config option from the server configuration directly instead getting from taskInfo.

Reply:
@jerqi It is different, this is a client side config, it configured by client, so I have to get it from shuffle TaskInfo.

Reply:
Maybe this config option should be same between server and client side.